addons/survey/controllers/main.py

```diff
@@ -357,19 +360,26 @@ def survey_submit(self, survey_token, answer_token, **post):
             # prevent cheating with users creating multiple 'user_input' before their last attempt
             return {}
 
-        page_id = int(post['page_id'])
-        questions = request.env['survey.question'].sudo().search([
-            ('survey_id', '=', survey_sudo.id),
-            ('page_id', '=', page_id)
-        ])
+        if survey_sudo.questions_layout == 'page_per_section':
```

Comment:
maybe we should extract this into a method on the `user_input` model to keep all logic business on model.